lib/ArrayCache.php

```diff
@@ -85,7 +88,12 @@ public function set(string $key, string $value, int $ttl = null): Promise {
         } else {
             throw new \Error("Invalid cache TTL ({$ttl}; integer >= 0 or null required");
         }
-
+        if (isset($this->sharedState->cache[$key])) {
+            unset($this->sharedState->cache[$key]);
+        }
+        if (count($this->sharedState->cache) >= $this->maxSize) {
```

Comment:
This can be `===`, because we only store one item at a time and pop one if it's too full. It needs a `$this->maxSize !== null &&`.

Reply:
But `null` convert to 0 during comparing

Reply:
With `===` it doesn't need the additional check, but with `>=` `null` doesn't mean unlimited, see https://3v4l.org/N7V31.

Reply:
You are completely right, seems I'm tired today